protocol/common/message_sender.go

```diff
@@ -427,16 +434,18 @@ func (s *MessageSender) sendCommunity(
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to decompress pubkey")
 		}
-		hash, newMessage, err = s.dispatchCommunityMessage(ctx, pubkey, payload, messageIDs, rawMessage.PubsubTopic)
+		hashes, newMessages, err = s.dispatchCommunityMessage(ctx, pubkey, payload, messageIDs, rawMessage.PubsubTopic)
 		if err != nil {
 			s.logger.Error("failed to send a community message", zap.Error(err))
 			return nil, errors.Wrap(err, "failed to send a message spec")
 		}
+	}
 
-		s.logger.Debug("sent community message ", zap.String("messageID", messageID.String()), zap.String("hash", types.EncodeHex(hash)))
+	for i, newMessage := range newMessages {
+		s.logger.Debug("sent community message ", zap.String("messageID", messageID.String()), zap.String("hash", types.EncodeHex(hashes[i])))
+		s.transport.Track(messageIDs, hashes[i], newMessage)
```

Comment:
Interesting, we probably need to update at some point what it means for a message to be dispatched received?

Reply:
Thanks, valid point. I will investigate it as followup: https://github.com/status-im/status-go/issues/4310